test/integration/features/addon-che.feature

```diff
@@ -0,0 +1,64 @@
+@addon-che @addon
+Feature: Che add-on
+  Che addon starts Eclipse Che
+
+  Scenario: Che add-on is part of Minishift
+     When executing "minishift addons list" succeeds
+     Then stdout should contain "che"
+  
+  Scenario: User starts Minishift
+    Given Minishift has state "Does Not Exist"
+     When executing "minishift start --memory=5GB" succeeds
+     Then Minishift should have state "Running"
+
+  Scenario: User applies Che add-on
+     When applying che addon with openshift token succeeds
+     Then stdout should contain "Please wait while the pods all startup!"
+
+  Scenario: Che is ready
+    Given Minishift has state "Running"
+     When executing "oc project mini-che" succeeds
+     Then service "che" rollout successfully within "300" seconds
+
+  Scenario: Che API is accessible
+     When user tries to get the che api endpoint
+     Then che api endpoint should not be empty
+
+  Scenario: Che stacks are accessible
+     When user tries to get the stacks information
+     Then the stacks should not be empty
+
+  Scenario Outline: User starts workspace, imports projects, checks run commands
+     When starting a workspace with stack "<stack>" succeeds
+     Then workspace should have state "RUNNING"
+
+     When importing the sample project "<sample>" succeeds
+     Then workspace should have 1 project
+
+     When user runs command on sample "<sample>"
+     Then exit code should be 0
+
+     When stopping a workspace succeeds
+     Then workspace should have state "STOPPED"
+
+     When workspace is removed
+     Then workspace removal should be successful
+
+    Examples:
+    | stack                 | sample                                                                   |
+    | Eclipse Vert.x        | https://github.com/openshiftio-vertx-boosters/vertx-http-booster         |
```

Comment:
@agajdosi I am not in favor of running all the stack here.

I will only allow one. We are not here to test all stacks probably we can create a separate CI system which will test all but not as part of core CI. I think `Java WildFly Swarm` or `Eclipse Vert.x` should be tested here which are more Red Hat product oriented. 
Testing all stack will increase the integration time dramatically. 
Already, we are reaching 3hrs timelimit :smile: .

Reply:
@budhram My suggestion is to run this only on nightly builds, if 3hrs limit is problem there, I would think of splitting cases to be run on 2 machines.

Reply:
@budhram Plan B: will take a look whether tagging examples is implemented in Godog, something like this: https://github.com/cucumber/cucumber-js/issues/305#issuecomment-74038183. If so, I can tag Wildfly and Eclipse as `quick` and the rest as `full` or something like that.

Reply:
@agajdosi  Solution from my side https://github.com/minishift/minishift/issues/2720